pkg/clickhouse/utils.go

```diff
@@ -9,12 +9,26 @@ import (
 )
 
 func GetDiskByPath(disks []Disk, dataPath string) string {
+	prefixLengthMap := map[int]string{}
+
 	for _, disk := range disks {
 		if strings.HasPrefix(dataPath, disk.Path) {
-			return disk.Name
+			prefixLengthMap[len(disk.Path)] = disk.Name
```

Comment:
When we have multiple disks with following map: disk1=/var/lib/clickhouse/disk1, disk2=/var/lib/clickhouse/disk2, disk3=/var/lib/clickhouse/disk3 then I'm not sure we will restore data properly. 
Could you add integration tests which cover this corner case?

Reply:
It has to restore as the default clickhouse paths looks like:
```metadata_path — Path on local FS to store metadata files for S3. Default value is /var/lib/clickhouse/disks/<disk_name>/.```

```data - var/lib/clickhouse/disks/<disk_name>/data```

See for[ more details.](https://clickhouse.tech/docs/en/engines/table-engines/mergetree-family/mergetree/)

I'll come up with intergration tests when I have time. But feel free to help if it's ASAP.